### PR TITLE
document pipeline

### PR DESCRIPTION
**Pull request by MarinPostma**
Document the `pipeline` HTTP API instead of the legacy one.

### CODE REVIEW

**Review by notrab — CHANGES_REQUESTED** [review]

File: sdk/http/reference.mdx [hunk 1]
```diff
@@ -27,99 +27,129 @@ Turso uses Bearer authentication, and requires your API token to be passed with
...
-        ":text": "Add a section for using parameters",
-        "$completed": 1
-      }
-    }
-  ]
+    "baton": "m7lVVgEvknpf1P1irxHsHqrAqH7BLiwO4DQIAwr93PdZWGvdBNugLSokSsCZNkry",
+    "requests": [
+        { "type": "execute", "stmt": { "sql": "CREATE TABLE users (name)" } },
+        { "type": "execute", "stmt": { "sql": "INSERT INTO users VALUES (\"adhoc\")" } },
+        { "type": "execute", "stmt": { "sql": "COMMIT } },
```
Comment (notrab) [inline sdk/http/reference.mdx:140]:
```suggestion
        { "type": "execute", "stmt": { "sql": "COMMIT" } },
```

File: sdk/http/reference.mdx [hunk 2]
```diff
@@ -27,99 +27,129 @@ Turso uses Bearer authentication, and requires your API token to be passed with
...
+    ]
 }
 ```
 
-</CodeGroup>
+We perform more queries in our transaction, commit, and then close the connection.
+
+Note that both transactions and connections have timeouts. Transaction have a 5 seconds window to complete, while connections get closed after 10 seconds of idle time.
+
+## Endpoints
```
Comment (notrab) [inline sdk/http/reference.mdx:150]: What do you think about moving the `pipeline` endpoint into this section, OR would you prefer to rename the sections here to be less about endpoints and more about obtaining version and checking health?

File: sdk/http/reference.mdx [hunk 3]
```diff
@@ -27,99 +27,129 @@ Turso uses Bearer authentication, and requires your API token to be passed with
...
+To perform a basic query, one can use the `POST /v2/pipeline` endpoint. This endpoint accepts a series of operation to perform against a Turso database connection. We are specifically interested in the following operations:
+- `execute`: execute a statement on the connection.
+- `close`: close the connection.
 
-<CodeGroup>
+By default a connection is left open until it timeouts. Every request on the connection bumps the timeout. Clients are expected to close their connection when it's not needed anymore.
 
-```json Request
+### Simple query
+Let's make a simple query:
```
Comment (notrab) [inline sdk/http/reference.mdx:41]:
```suggestion
```

File: sdk/http/reference.mdx (hunk 2 above)
Comment (MarinPostma) [reply sdk/http/reference.mdx:150]: I like option 2 better. Do you have suggestions for a section title?

Comment (notrab) [reply sdk/http/reference.mdx:150]: The idea of the Reference was to be less walkthrough and more "reference-y". The Quickstart could be a great place to put the current usage content you put together, and then the Reference can mention only the input/output spec.

Is that ok with you?

**Pull request merged**